src/ngx_http_lua_socket_tcp.c

```diff
@@ -2673,6 +2694,111 @@ ngx_http_lua_socket_tcp_settimeout(lua_State *L)
 }
 
 
+static int
+ngx_http_lua_socket_tcp_set_receive_timeout(lua_State *L)
+{
+    int                     n;
+    ngx_int_t               timeout;
+
+    ngx_http_lua_socket_tcp_upstream_t  *u;
+
+    n = lua_gettop(L);
+
+    if (n != 2) {
+        return luaL_error(L, "ngx.socket set_receive_timeout: expecting at least 2 "
+                          "arguments (including the object) but seen %d",
+                          lua_gettop(L));
+    }
+
+    timeout = (ngx_int_t) lua_tonumber(L, 2);
+
+    lua_rawseti(L, 1, SOCKET_TIMEOUT_INDEX);
```

Comment:
@aviramc Nope. We already have `receive` and `receiveuntil` methods and we may add more reading methods like those in the future. Don't want to give the user the impression that this timeout is only for the `receive()` method.

Also, we already have the `lua_socket_read_timeout` directive BTW :)